x/supplier/client/cli/stake_options_reader.go

```diff
@@ -0,0 +1,157 @@
+package cli
+
+import (
+	"fmt"
+	"net/url"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"gopkg.in/yaml.v2"
+
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// ParsedStakeConfig is a parsed version of the stake config file
+type ParsedStakeConfig struct {
+	stake    sdk.Coin
+	services []*sharedtypes.SupplierServiceConfig
+}
+
+// StakeConfig is the structure of the stake config file
+type StakeConfig struct {
+	Stake    string         `yaml:"stake"`
+	Services []StakeService `yaml:"services"`
+}
+
+// StakeService is the structure describing a single service stake entry in the stake config file
+type StakeService struct {
+	ServiceId string            `yaml:"service_id"`
+	Endpoints []ServiceEndpoint `yaml:"endpoints"`
+}
+
+// ServiceEndpoint is the structure describing a single service endpoint in the stake config file
+type ServiceEndpoint struct {
+	Url     string            `yaml:"url"`
+	RPCType string            `yaml:"rpc_type"`
+	Config  map[string]string `yaml:"config"`
+}
+
+// parseStakeConfig parses the stake config file into a parsedStakeConfig
+func parseStakeConfig(configContent []byte) (*ParsedStakeConfig, error) {
+	var (
+		stakeConfig *StakeConfig
+		err         error
+	)
+
+	// Unmarshal the stake config file into a stakeConfig
+	if err = yaml.Unmarshal(configContent, &stakeConfig); err != nil {
+		return nil, err
+	}
+
+	// Prepare the parsedStakeConfig
+	parsedStakeConfig := &ParsedStakeConfig{}
+
+	// Parse the stake amount and assign it to the parsedStakeConfig
+	parsedStakeConfig.stake, err = sdk.ParseCoinNormalized(stakeConfig.Stake)
+	if err != nil {
+		return nil, err
+	}
+
+	// Prepare the services slice
+	var services []*sharedtypes.SupplierServiceConfig
+
+	// Populate the services slice
+	for _, svc := range stakeConfig.Services {
+		// Validate the serviceId
+		// TODO_TECH_DEBT: This should be validated against some governance state
```

Comment:
1. NIT: We generally use `TODO_TECHDEBT`
2. Please use `sharedhelpers.IsValidService`. Searching for it in the codebase should bring up examples.